cleverhans/utils_eager.py

```diff
@@ -0,0 +1,252 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from distutils.version import LooseVersion
+import math
+import numpy as np
+import os
+from six.moves import xrange
+import tensorflow as tf
+import time
+import warnings
+
+from .utils import batch_indices, _ArgsWrapper, create_logger
+
+_logger = create_logger("cleverhans.utils.eager")
+
+
+def model_loss(logits, labels, mean=True):
+    """
+    Loss Definiton
+    :param labels: correct labels
+    :param logits: output of the model
+    :param mean: boolean indicating whether should return mean of loss
+                 or vector of losses for each input of the batch
+    :return: return mean of loss if True, otherwise return vector with per
+             sample loss
+    """
+    try:
+        labels = tf.stop_gradient(labels)
+        out = tf.nn.softmax_cross_entropy_with_logits_v2(
+                                                logits=logits, labels=labels)
+    except AttributeError:
+        out = tf.nn.softmax_cross_entropy_with_logits(
+                                                logits=logits, labels=labels)
+
+    if mean:
+        out = tf.reduce_mean(out)
+    return out
+
+
+def model_train(model, X_train=None, Y_train=None, save=False,
```

Comment:
This function follows the deprecated mode, it does not use a loss. Check the function train, which replaces it.

Reply:
Ohh, this change was done, after this PR is submitted, that is not an issue, i can update this accordingly.